app/controllers/rebate_forms_controller.rb

```diff
@@ -42,6 +44,16 @@ def rebate_form_params
               :council,
               :total_rates,
               :location,
-              fields: {})
+              fields: {},
+              applicant_signature: {},
+              witness_signature: {})
+  end
+
+  def decode_jwt(token)
+    JWT.decode token, ENV['HMAC_SECRET'], true, algorithm: 'HS256'
+  end
+
+  def correct_token(decoded_token)
+    raise JsonapiCompliable::Errors::RecordNotFound unless decoded_token['per'] == 'fetch_application_and_submit_signatures'
```

Comment:
We could save some bytes in the generated qr by renaming `per` to something like 'fetch_sign'.